core/store/src/db.rs

```diff
@@ -126,8 +128,12 @@ pub struct RocksDB {
 unsafe impl Send for RocksDB {}
 unsafe impl Sync for RocksDB {}
 
-fn col_name(col: DBCol) -> String {
-    format!("col{}", col as usize)
+struct ColName(DBCol);
```

Comment:
This struct feels like a needless abstraction. It raises questions, because `impl std::convert::Into<String> for ColName` is very un-idiomatic: it should be `From`, not `Into`, and it really should just be `Display`.

I think we need this becaues `Into<String>` is what `ColumnFamilyDescriptor::new(`. 

But the reason why it does that is "ergonomics", just making sure that `ColumnFamilyDescriptor::new` with either `String` or `&str`: interally, it immediately converts the thing to String

So here are we effectively introducing some abstract contraption because authros of rocksdb wanted to make it more ergonomic for us :rofl: 

I'd suggest maybe just 

```rust
impl DBCol {
  pub(crate) fn rocksdb_col_name(&self) -> String
}
```

---

Alternatively, do we really need this to be different from `col.to_string()`? Maybe let's adjust `impl fmt::Display for DBCol` to be reasonable identifiers without spaces, rather than descriptions, and use those when openeing DB? 

Also: `ColumnFamilyDescriptor::new` should just take `String`, `AsRef` / `Into` polymorphism is very rarely a good idea

Reply:
You’re right.  I’m not sure why I was thinking that .into() call inside ColumnFamilyDescriptor::new would be a clone.

> it's part of DB schema, so its not really private

That’s not necessarily true.  It’s part of the RocksDB schema but it’s not part of the near_store.  TestDB for example does not use col_name.  And if we move to having cloud storage for archival data, that wouldn’t necessarily use col_name either.  (Regardless, I’m gonna merge this PR; we can move the fn in another PR).

Reply:
No strong opinion on whether to move or not!